packages/tools/accessibility/src/HtmlTwin/htmlTwinItemAdapter.tsx

```diff
@@ -0,0 +1,117 @@
+    if (node instanceof Node) {
+        return new HTMLTwinNodeItem(node, scene);
+    } else {
+        return new HTMLTwinGUIItem(node, scene);
+    }
+}
+
+export function HTMLTwinItemAdapter(props: { node: AccessibilityEntity; scene: Scene; options: IHTMLTwinRendererOptions }) {
+    const { node, scene, options } = props;
+    if (!node) {
+        return null;
+    }
+    const [twinItem, setTwinItem] = useState<HTMLTwinItem>(getTwinItemFromNode(node, scene));
+    useEffect(() => {
+        setTwinItem(getTwinItemFromNode(node, scene));
+    }, [node]);
+
+    const [isVisibleState, setIsVisibleState] = useState(isVisible(props.node));
+    useEffect(() => {
+        const observable = (node as any).onEnabledStateChangedObservable;
+        const observer = observable.add((value: boolean) => {
+            setIsVisibleState(value);
+        });
+        return () => {
+            observable.remove(observer);
+        };
+    }, [node]);
+
+    const sceneContext = useContext(SceneContext);
+    useEffect(() => {
+        const observable = (node as any).onDisposeObservable;
+        const observer = observable.add(() => {
+            sceneContext.updateScene();
+        });
+        return () => {
+            observable.remove(observer);
+        };
+    }, [node]);
+
+    const [description, setDescription] = useState(twinItem?.getDescription(options));
+    useEffect(() => {
+        const observable = node.onAccessibilityTagChangedObservable;
+        const observer = observable.add(() => {
+            setDescription(twinItem?.getDescription(options));
+        });
+        return () => {
+            observable.remove(observer);
+        };
+    }, [node]);
+
+    useEffect(() => {
+        setDescription(twinItem?.getDescription(options));
+    }, [twinItem]);
+
+    const [children, setChildren] = useState(getDirectChildrenOf(props.node));
+    if (node instanceof Container) {
+        useEffect(() => {
+            const observable = node.onControlAddedObservable;
+            const observer = observable.add(() => {
+                setChildren([...getDirectChildrenOf(props.node)]);
+            });
+            return () => {
+                observable.remove(observer);
+            };
+        }, [node]);
+        useEffect(() => {
+            const observable = node.onControlRemovedObservable;
+            const observer = observable.add(() => {
+                setChildren([...getDirectChildrenOf(props.node)]);
+            });
+            return () => {
+                observable.remove(observer);
+            };
+        }, [node]);
+    }
+    if (node instanceof Control) {
+        useEffect(() => {
+            const observable = node.onIsVisibleChangedObservable;
+            const observer = observable.add(() => {
+                setIsVisibleState(isVisible(props.node));
+            });
+            return () => {
+                observable.remove(observer);
+            };
+        }, [node]);
+    }

```

Comment:
useEffect should not be conditional. You can check for the instanceof inside the useEffect callback instead.